### PR TITLE
fix(xsnap): orderly fail-stop on heap exhaustion

### PR DESCRIPTION
**Pull request by dckc**
Note https://github.com/Moddable-OpenSource/moddable/issues/567 is only fixed for certain cases.

### CODE REVIEW

File: packages/xsnap/test/test-xsnap.js [hunk 1]
```diff
@@ -349,3 +349,17 @@ test('normal close of pathological script', async t => {
...
 });
+
+test('heap exhaustion: orderly fail-stop', async t => {
+  const grow = `
+  let stuff = '1234';
+  while (true) {
+    stuff = stuff + stuff;
+  }
+  `;
+  for await (const debug of [false, true]) {
```
Comment (michaelfig) [inline packages/xsnap/test/test-xsnap.js]: No need for `for await...of` here.  This isn't an async iterable.

```suggestion
  for (const debug of [false, true]) {
```

**Review by michaelfig — APPROVED** [review]
LGTM.

File: packages/xsnap/test/test-xsnap.js (hunk 1 above)
Comment (dckc) [reply packages/xsnap/test/test-xsnap.js]: I got a red squiggle when I wrote it that way... I thought it was a correctness thing, but now I see it's https://eslint.org/docs/rules/no-await-in-loop ... that recommends `Promise.all`. hm. I guess there's no data dependency here, but intermixed logs can be confusing. I suppose that argues for splitting this into 2 tests to let ava manage the parallelism and correlating logs. hm. Maybe next time...

**Review by warner — APPROVED** [review]
looks great

**Pull request merged**